test/test_client.py

```diff
@@ -474,6 +481,67 @@ def test_set_plc_datetime(self):
         # Can't actual set datetime in emulated PLC, get_plc_datetime always returns system time.
         # self.assertEqual(new_dt, self.client.get_plc_datetime())
 
+    def test_wait_as_completion_pass(self, timeout=10):
+        # Cli_WaitAsCompletion
+        # prepare Server with values
+        area = snap7.types.areas.DB
+        dbnumber = 1
+        size = 1
+        start = 1
+        data = bytearray(size)
+        self.client.write_area(area, dbnumber, start, data)
+        # start as_request and test
+        p_data = self.client.as_db_read(dbnumber, start, size)
+        self.client.wait_as_completion(timeout)
+        self.assertEqual(bytearray(p_data), data)
+
+    def test_wait_as_completion_timeouted(self, timeout=0, tries=500):
+        # Cli_WaitAsCompletion
+        # prepare Server
+        area = snap7.types.areas.DB
+        dbnumber = 1
+        size = 1
+        start = 1
+        data = bytearray(size)
+        self.client.write_area(area, dbnumber, start, data)
+        # start as_request and wait for zero seconds to try trigger timeout
+        for i in range(tries):
+            self.client.as_db_read(dbnumber, start, size)
+            try:
+                self.client.wait_as_completion(timeout)
+            except Snap7Exception as s7_err:
+                if not s7_err.args[0] == b'CLI : Job Timeout':
+                    self.fail(f"While waiting another error appeared: {s7_err}")
+                return
```

Comment:
why do you return here? shouldn't it just continue?

Reply:
Otherwise I would stuck in the loop, testing the behaviour multiple times(which is unnecessarry) and then, if the loop would be done, the test would fail.

So if once a Job Timeout was raised correctly, the test shall pass (via return).

Reply:
ah yes, skimmed too quickly.